### ISSUE TITLE
A question about agrument "fb"

### ISSUE THREAD

**Issue by syuie** [valdersoul/bn-vae#2]
Very appreciate your work on keeping kl vanishing away and thank you for your code!
I don't understand whether fb == 1 or fb == 2 represents objective with free bits within the code below(in file text_anneal_fb.py):
```
if args.fb == 0 or args.fb == 4:
    loss, loss_rc, loss_kl = vae.loss(batch_data, kl_weight, nsamples=args.nsamples)
elif args.fb == 1:
    loss, loss_rc, loss_kl = vae.loss(batch_data, kl_weight, nsamples=args.nsamples)
    kl_mask = (loss_kl > args.target_kl).float()
    loss = loss_rc + kl_mask * kl_weight * loss_kl 
elif args.fb == 2:
    mu, logvar = vae.encoder(batch_data)
    z = vae.encoder.reparameterize(mu, logvar, args.nsamples)
    loss_kl = 0.5 * (mu.pow(2) + logvar.exp() - logvar - 1)
    kl_mask = (loss_kl > dim_target_kl).float()
    fake_loss_kl = (kl_mask * loss_kl).sum(dim=1)
    loss_rc = vae.decoder.reconstruct_error(batch_data, z).mean(dim=1)
    loss = loss_rc + kl_weight * fake_loss_kl 
elif args.fb == 3:
    loss, loss_rc, loss_kl = vae.loss(batch_data, kl_weight, nsamples=args.nsamples)
```
For my understanding, when fb == 2, loss = loss_rc + max(target_kl, loss_kl), and when fb == 1, loss = lossrc + (0 if kl < target_kl else kl). If the case fb == 2 represents objective with free bits, then what does the case fb == 1 represent?

**Comment by valdersoul** [comment]
Hi,

in fb==1, the freebits is set in averaged KL across dimension. 
in fb==2, the freebits is set for KL with individual dimension.

**Comment by syuie** [comment]
Understood, thank you for your reply!
